controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java

```diff
@@ -432,6 +432,16 @@ public void handle(long term, MemberState.Role role) {
                                     DLedgerController.this.startScheduling();
                                     break;
                                 }
+                                if (!DLedgerController.this.getMemberState().isLeader()) {
+                                    // now is not a leader
+                                    log.error("Append a initial log failed because current state is not leader");
+                                    break;
+                                }
+                                log.error("Controller leader append initial log failed, try again");
+                                tryTimes++;
+                                if (tryTimes % 3 == 0) {
+                                    log.warn("Controller leader append initial log failed too many times, please wait a while");
+                                }
                             } catch (final Throwable e) {
                                 log.error("Error happen when controller leader append initial request to dledger", e);
                                 tryTimes++;
```

Comment:
If an exception is thrown, do we need to determine whether it is a leader?

Reply:
Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)

Reply:
> Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)

How about creating a function to deal with it？

Reply:
> > Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)
> 
> How about creating a function to deal with it？

got it!

Reply:
> > Yep~ Maybe we can check the state once the appending is failed(appendToDLedfer return false or appendToDLedger AndWait throws an exception)
> 
> How about creating a function to deal with it？

I find that maybe it is uncessary to add a function to deal with it.